exporter/signalfxexporter/translation/translator.go

```diff
@@ -266,6 +344,86 @@ func (mp *MetricTranslator) TranslateDimension(orig string) string {
 	return orig
 }
 
+// aggregateDatapoints aggregates datapoints assuming that they have
+// the same Timestamp, MetricType, Metric and Source fields.
+func aggregateDatapoints(
+	logger *zap.Logger,
+	dps []*sfxpb.DataPoint,
+	dimensionKey string,
+	aggregation AggregationMethod,
+) []*sfxpb.DataPoint {
+	if len(dps) == 0 {
+		return nil
+	}
+
+	// group datapoints by dimension values
+	dimValToDps := make(map[string][]*sfxpb.DataPoint, len(dps))
+	for i, dp := range dps {
+		var dimensionFound bool
+		for _, d := range dp.Dimensions {
+			if d.Key == dimensionKey {
+				if _, ok := dimValToDps[d.Value]; !ok {
+					dimValToDps[d.Value] = make([]*sfxpb.DataPoint, 0, len(dps)-i)
+				}
+				dimValToDps[d.Value] = append(dimValToDps[d.Value], dp)
+				dimensionFound = true
+			}
+		}
+		if !dimensionFound {
+			logger.Debug("dimension to aggregate by is not found, datapoint is dropped",
+				zap.String("metric", dp.Metric), zap.String("dimension", dimensionKey))
+		}
+	}
+
+	// Get aggregated results
+	result := make([]*sfxpb.DataPoint, 0, len(dimValToDps))
+	for _, dps := range dimValToDps {
+		dp := proto.Clone(dps[0]).(*sfxpb.DataPoint)
+		dp.Dimensions = getCommonDimensions(dps)
```

Comment:
Why do we need to calculate common dimensions? Aren't we reducing this metric to only contain the single dimension across which we aggregate?

Reply:
yes, this is a bit unusual for aggregations, but since we are getting a bunch of common dimensions from resource attributes like k8s.node.name, k8s.cluster.name etc, we would like to keep them as is.

Reply:
Instead , we could explicitly specify list of dimensions to aggregate across

Reply:
I'm changing it to use this approach ^ . Please do not review yet

Reply:
@tigrannajaryan I refactored it accordingly. PTAL